### PR TITLE
TL/NCCL: use memops to check completion

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Use cudastreamwritevalue to signal completion of nccl collective. If it's enabled to progress is needed for TL NCCL

### CODE REVIEW

File: src/components/tl/nccl/tl_nccl_coll.c [hunk 1]
```diff
@@ -57,13 +57,29 @@ static inline ucc_status_t ucc_nccl_check_dt_supported(ucc_datatype_t dt1,
...
+                                         cudaStream_t stream)
 {
-    ucc_tl_nccl_task_t *task = ucc_derived_of(coll_task, ucc_tl_nccl_task_t);
-    ucc_status_t status;
-
-    status = ucc_mc_ee_event_test(task->completed, UCC_EE_CUDA_STREAM);
-    coll_task->super.status = status;
+    ucc_tl_nccl_context_t *ctx  = ucc_derived_of(task->team->super.super.context,
+                                                 ucc_tl_nccl_context_t);
+    ucc_status_t status = UCC_OK;
```
Comment (vspetrov) [inline src/components/tl/nccl/tl_nccl_coll.c:65]: align on = and align cu_status with status; empty line after local vars

File: src/components/tl/nccl/tl_nccl_context.c [hunk 2]
```diff
@@ -5,11 +5,68 @@
...
+    .obj_cleanup   = NULL
+};
+
+static ucc_status_t ucc_tl_nccl_req_mapped_mpool_chunk_malloc(ucc_mpool_t *mp,
+                                                              size_t *size_p,
+                                                              void ** chunk_p)
+{
+    ucc_status_t status;
+
+    // TODO check for error
```
Comment (vspetrov) [inline src/components/tl/nccl/tl_nccl_context.c:42]: plz fix TODO

**Review by vspetrov — APPROVED** [review]

File: src/components/tl/nccl/tl_nccl_context.c [hunk 3]
```diff
@@ -5,11 +5,71 @@
...
+                                                  void *chunk)
+{
+    ucc_tl_nccl_task_t *req = (ucc_tl_nccl_task_t*) obj;
+    cudaError_t st;
+    st = cudaHostGetDevicePointer((void**)(&req->dev_status),
+                                  (void *)&req->super.super.status, 0);
+    if (st != cudaSuccess) {
+        req->super.super.status = UCC_ERR_NO_MESSAGE;
+    }
+    req->super.progress = NULL;
```
Comment (bureddy) [inline src/components/tl/nccl/tl_nccl_context.c:66]: progress is NULL, So, when does finalize() called in case of mem_ops?

Comment (Sergei-Lebedev) [reply src/components/tl/nccl/tl_nccl_context.c:66]: yes, progress is intentionally NULL, custreamwritevalue will update status and after that user might call finalize

**Review by bureddy — APPROVED** [review]

**Pull request merged**